File: 3603/3603.c
int minimumCostPathWithAlternating(struct TreeNode* root) {
    if (!root) return 0;
    int l = minimumCostPathWithAlternating(root->left);
    int r = minimumCostPathWithAlternating(root->right);
    return 1 + (l > r ? l : r);
}

File: 3603/3603.go
func minimumCostPathWithAlternating(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := minimumCostPathWithAlternating(root.Left)
	r := minimumCostPathWithAlternating(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}